crates/api/src/site/registration_applications/tests.rs

```diff
@@ -0,0 +1,404 @@
+  expected_unread_applications_len += 1;
+  expected_applications_len += 1;
+
+  let (application_count, unread_applications, all_applications) =
+    get_application_statuses(&context, admin_local_user_view.clone()).await?;
+
+  // When email verification is required and the email is verified the application should be
+  // visible to admins
+  assert_eq!(
+    expected_application_count,
+    application_count.registration_applications
+  );
+  assert_eq!(
+    expected_unread_applications_len,
+    unread_applications.registration_applications.len()
+  );
+  assert!(
+    !unread_applications.registration_applications[0]
+      .creator_local_user
+      .accepted_application
+  );
+  assert_eq!(
+    expected_applications_len,
+    all_applications.registration_applications.len()
+  );
+
+  let approval = approve_registration_application(
+    Json(ApproveRegistrationApplication {
+      id: app_with_email.id,
+      approve: true,
+      deny_reason: None,
+    }),
+    context.reset_request_count(),
+    admin_local_user_view.clone(),
+  )
+  .await;
+  // Approval should be processed up until email sending is attempted
+  assert!(approval.is_err_and(|e| e.error_type == LemmyErrorType::NoEmailSetup));
+
+  expected_application_count -= 1;
+  expected_unread_applications_len -= 1;
+
+  let (application_count, unread_applications, all_applications) =
+    get_application_statuses(&context, admin_local_user_view.clone()).await?;
+
+  // When the application is approved it should only be returned for unread queries
+  assert_eq!(
+    expected_application_count,
+    application_count.registration_applications
+  );
+  assert_eq!(
+    expected_unread_applications_len,
+    unread_applications.registration_applications.len()
+  );
+  assert_eq!(
+    expected_applications_len,
+    all_applications.registration_applications.len()
+  );
+  assert!(
+    all_applications.registration_applications[0]
+      .creator_local_user
+      .accepted_application
+  );
+
+  signup(
+    pool,
+    instance.id,
+    "user_w_email_2",
+    Some("lemmy2@localhost"),
+  )
+  .await?;
+  let (application_count, unread_applications, all_applications) =
+    get_application_statuses(&context, admin_local_user_view.clone()).await?;
+
+  // Email not verified, so application still not visible
+  assert_eq!(
+    expected_application_count,
+    application_count.registration_applications
+  );
+  assert_eq!(
+    expected_unread_applications_len,
+    unread_applications.registration_applications.len()
+  );
+  assert_eq!(
+    expected_applications_len,
+    all_applications.registration_applications.len()
+  );
+
+  update_site(
+    Json(EditSite {
+      require_email_verification: Some(false),
+      ..Default::default()
+    }),
+    context.reset_request_count(),
+    admin_local_user_view.clone(),
+  )
+  .await?;
+
+  // TODO: There is probably a better way to ensure cache invalidation
+  tokio::time::sleep(CACHE_DURATION_API).await;
```

Comment:
This test calls some web api implementations, but site settings like the application requirement is cached:
https://github.com/LemmyNet/lemmy/blob/128e78f7c22279fec27d3213a05b440cb1f8db59/crates/db_schema/src/impls/local_site.rs#L21-L26
The timeout isn't too long at just one second, so it's tolerable, but I'm wondering if there is some way to invalidate the cache?